packages/reactive-element/src/decorators/custom-element.ts

```diff
@@ -13,6 +13,125 @@
 
 import type {Constructor} from './base.js';
 
+declare global {
+  interface CustomElementRegistry {
+    getName: (ctor: CustomElementConstructor) => string | null;
+  }
+}
+
+export type DefineErrorCallback = (
+  error: Error,
+  tagName: string,
+  ctor: CustomElementConstructor
```

Comment:
I think tag name and constructor should be on the error object. We can make an Error subclass for this.

We also need both the existing and new constructors on the error so that handlers can introspect them for compatibility.